### PR TITLE
Version 1.0

### PR DESCRIPTION
**Pull request by JYCabello**
Added a XML docs for many of the functions, will do more in another push.

Bumped version to 1.0

### CODE REVIEW

**Review by theonlyroboboro — APPROVED** [review]

File: src/DeFuncto.Core/ActivePatternMatching/ActivePattern.cs [hunk 1]
```diff
@@ -3,30 +3,96 @@
 
 namespace DeFuncto.ActivePatternMatching;
 
+/// <summary>
+/// Base class for the active pattern matching.
+/// This bit is not, and never will, be documented on the official documentation.
```
Comment (aralmo) [inline src/DeFuncto.Core/ActivePatternMatching/ActivePattern.cs:8]: I would move the part that talks about jumping to f# out of the summary, as a remark maybe.

File: src/DeFuncto.Core/Extensions/Tasks.cs [hunk 2]
```diff
@@ -5,35 +5,86 @@
...
+    /// <param name="self">The stacked task.</param>
+    /// <typeparam name="T">The type of the value in the innermost.</typeparam>
+    /// <returns> The flattened task. </returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static async Task<T> Flatten<T>(this Task<Task<T>> self) =>
         await await self;
 
-    [Pure]
+    /// <summary>
+    /// Flattens an array of lists inside a task.
```
Comment (aralmo) [inline src/DeFuncto.Core/Extensions/Tasks.cs:59]:
```suggestion
    /// Asynchronously reduces an array of lists into a one dimension list.
```

File: src/DeFuncto.Core/Extensions/Tasks.cs [hunk 3]
```diff
@@ -56,61 +107,134 @@ async Task<T> WrapInSemaphore(Func<Task<T>> f)
...
+    /// <typeparam name="TOut">Type of the elements of the output.</typeparam>
+    /// <returns>The new enumerable, in a task.</returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static Task<IEnumerable<TOut>> Select<TIn, TOut>(this Task<IEnumerable<TIn>> self, Func<TIn, TOut> func) =>
         self.Map(ienumerable => ienumerable.Select(func));
 
-    [Pure]
+    /// <summary>
+    /// Performs a select in an List returned by a task.
+    /// Might seem redundant since there's already an IEnumerable version, but the type inference in C#
```
Comment (aralmo) [inline src/DeFuncto.Core/Extensions/Tasks.cs:176]:
```suggestion
```
I would either move this observation to a remark or regullar comment or remove it.

File: src/DeFuncto.Core/Extensions/Tasks.cs [hunk 4]
```diff
@@ -56,61 +107,134 @@ async Task<T> WrapInSemaphore(Func<Task<T>> f)
...
+    /// is not really friendly in this case.
```
Comment (aralmo) [inline src/DeFuncto.Core/Extensions/Tasks.cs:177]:
```suggestion
```

File: src/DeFuncto.Core/Extensions/Tasks.cs [hunk 5]
```diff
@@ -56,61 +107,134 @@ async Task<T> WrapInSemaphore(Func<Task<T>> f)
...
 
-    [Pure]
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static Task<IEnumerable<TResult>> Select<TIn, TResult>(this Task<TIn[]> self, Func<TIn, TResult> mapper) =>
-        self.Map(x => x.Select(mapper));
-
-    [Pure]
+    /// <summary>
+    /// Performs a select in an Array returned by a task.
+    /// Might seem redundant since there's already an IEnumerable version, but the type inference in C#
```
Comment (aralmo) [inline src/DeFuncto.Core/Extensions/Tasks.cs:190]:
```suggestion
```

File: src/DeFuncto.Core/Extensions/Tasks.cs [hunk 6]
```diff
@@ -56,61 +107,134 @@ async Task<T> WrapInSemaphore(Func<Task<T>> f)
...
+    /// is not really friendly in this case.
```
Comment (aralmo) [inline src/DeFuncto.Core/Extensions/Tasks.cs:191]:
```suggestion
```

**Review by aralmo — COMMENTED** [review]
Added some suggestions on comments based on personal practices.

File: src/DeFuncto.Core/Extensions/Nullables.cs [hunk 7]
```diff
@@ -2,13 +2,32 @@
...
+    /// <typeparam name="T">Input type.</typeparam>
+    /// <typeparam name="TReturn">Projected type.</typeparam>
+    /// <returns>Projected value, or null.</returns>
     public static TReturn? Map<T, TReturn>(this T? self, Func<T, TReturn> f)
         where T : struct
         where TReturn : struct =>
         self is null ? null : f(self.Value);
 
+    /// <summary>
+    /// Binds a nullable value.
```
[the remark this answers is not in the document]
Comment (JYCabello) [reply src/DeFuncto.Core/Extensions/Nullables.cs:24]: No, this is a binding, projects to a nullable.

File: src/DeFuncto.Core/Extensions/Functions.cs [hunk 8]
```diff
@@ -54,13 +93,25 @@ public static class Functions
             return unit;
         };
 
+    /// <summary>
+    /// Turns an async unit returning function into an asynchronous action.
```
[the remark this answers is not in the document]
Comment (JYCabello) [reply src/DeFuncto.Core/Extensions/Functions.cs:97]: Done

File: src/DeFuncto.Core/ActivePatternMatching/ActivePattern.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JYCabello) [reply src/DeFuncto.Core/ActivePatternMatching/ActivePattern.cs:8]: Done

File: src/DeFuncto.Core/Extensions/Tasks.cs [hunk 9]
```diff
@@ -5,35 +5,86 @@
...
+    /// <returns>
+    /// A task that will return the result of computing the original task's result with the
+    /// projection function and flattening with the wrapping task.
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Task<TOut> Map<TIn, TOut>(this Task<TIn> self, Func<TIn, Task<TOut>> f) =>
+        self.Map<TIn, Task<TOut>>(f).Flatten();
+
+    /// <summary>
+    /// Flattens a stacked task.
```
[the remark this answers is not in the document]
Comment (JYCabello) [reply src/DeFuncto.Core/Extensions/Tasks.cs:49]: It's not a reduction.

**Review by aralmo — APPROVED** [review]

**Comment by aralmo** [comment]
it's alive!

**Pull request merged**